Main

Mike234134 opened the pull request:
my solution

Oluwatobi02 commented:
Test your code before you push. use the tests below as a guide to test your solution.

targetDate = "2024-08-13-12-00-00".split('-')
datesList = [
    "2024-08-13-11-45-00".split('-'),
    "2024-08-13-12-45-00".split('-'),
    "2024-08-13 13-00-00".split('-')
]
    
checkNearbyDates(targetDate, datesList)

Mike234134 commented:
check again  def checkNearbyDates(targetDate,datesList):
    date=targetDate[3:4]
    print(date)
    #finding the hour of the hours
    newdate=date[0]
    #since the hour is still a list we index into it again to get the number
    newdate = int(newdate)*60
    
    
    

    for i in datesList:
        newdatesListnumber= i[3:4]
        newst=newdatesListnumber[0]
        newstdate=int(newst)*60
        timediff = newstdate-newdate
    if timediff in range(1,31):
        print(True)
    else:
        print(False)
        
   
checkNearbyDates( "2024-08-13-12-00-00".split('-'), [
    "2024-08-13-11-45-00".split('-'),
    "2024-08-13-12-45-00".split('-'),
    "2024-08-13-13-00-00".split('-')])
